modules/extensions/src/terrain/terrain-cover.ts

```diff
@@ -0,0 +1,251 @@
+import {Framebuffer} from '@luma.gl/core';
+
+import {WebMercatorViewport, OrthographicViewport} from '@deck.gl/core';
+import type {Layer, Viewport} from '@deck.gl/core';
+
+import {createRenderTarget} from './utils';
+
+const unitGeospatialViewport = new WebMercatorViewport({width: 1, height: 1});
+const unitNonGeospatialViewport = new OrthographicViewport({width: 1, height: 1});
+
+const MAX_SIZE = 2048;
+
+// TODO - import from loaders when Tileset2D is split out
+type GeoBoundingBox = {west: number; north: number; east: number; south: number};
+type NonGeoBoundingBox = {left: number; top: number; right: number; bottom: number};
+type Tile2DHeader = {
+  bbox: GeoBoundingBox | NonGeoBoundingBox;
+};
+
+/** Class to manage draped texture for each terrain layer */
+export default class TerrainCover {
+  private fbo?: Framebuffer;
+  private pickingFbo?: Framebuffer;
+  private zoom: number = 0;
+  private bounds: [number[], number[]] | null = null;
+  private layers: string[] = [];
+  private tile: Tile2DHeader | null;
+
+  isDirty: boolean = true;
+  owner: Layer;
+  renderViewport: Viewport | null = null;
+  commonBounds: [number[], number[]] | null = null;
+
+  constructor(owner: Layer) {
+    this.owner = owner;
+    this.tile = getTile(owner);
+  }
+
+  get id() {
+    return this.owner.id;
+  }
+
+  get isActive(): boolean {
+    const owner = this.owner.getCurrentLayer();
+    return owner !== null && !owner.lifecycle.startsWith('Finalized');
```

Comment:
Raking through the lifecycle string really breaks the abstraction. Would it make sense to add a layer method for this?

```suggestion
    return layer.isActive();
```

What would the documentation / exact semantics of that method be?

Reply:
A layer is currently in use (initialized and not finalized). Examples:

- A layer instance constructed but never added to deck's layer stack - returns `false`
- A layer instance added to deck's layer stack, and later matched by new versions of the same layer - returns `true`
- A layer instance added to deck's layer stack, and later removed - returns `false`
- A layer instance added to deck's layer stack, then removed, and later added back - returns `false`